graph_many.py

```diff
@@ -0,0 +1,91 @@
+import re
+import sys
+import collections
+try:
+    import numpy as np 
+except ImportError:
+    print('You need to install numpy.')
+    sys.exit(1)
+try:
+    import plotly.express as px
+except ImportError:
+    print('You need to install plotly.express.')
+    sys.exit(1)
+try:
+    import kaleido
+except ImportError:
+    print('You need to install kaleido.')
+    sys.exit(1)
+
+if len(sys.argv) != 2:
+    print('Usage: %s results.txt' % sys.argv[0])
+    print('Where results.txt is the output of the benchmark script. I.e.')
+    print(' mimalloc-bench/out/bench/benchres.csv')
+    print()
+    print('The script is designed to visualise the results of running the main benchmark')
+    print('script with multiple runs for each benchmark. For example running:')
+    print('  ../../bench.sh -r=10 mi sn allt')
+    print('from the mimalloc-bench/out/bench directory.')
+    print()
+    print('The script will create two graphs, one for time and one for memory.')
+    print('The graphs will be saved as time.html, time.png and time.pdf, and')
+    print('memory.html, memory.png and memory.pdf.')
+    print()
+    print('The graphs will be normalised by the mean time/memory for each benchmark.')
+    print('The normalised time/memory uses a log scale.')
+    sys.exit(1)
+
+r = re.compile('^([^ ]+) +([^ ]+) +([0-9:.]+) +([0-9]+)')
+data = []
+test_names = set()
+
+# read in the data
+with open(sys.argv[1]) as f:
+    for l in f.readlines():
+        match = r.search(l)
+        if not match:
+            continue
+        test_name = match.group(1)
+        alloc_name = match.group(2)
+        time_split = match.group(3).split(':')
+        memory = match.group(4)
```

Comment:
I think™ that this might work:

```suggestion
        _, test_name, alloc_name, time_split, memory = match
        time_split = time_split.split(':')
```